src/main/java/nextstep/courses/domain/Session.java

```diff
@@ -17,21 +22,35 @@ public class Session {
 
     private final SessionType sessionType;
 
+    private final SessionUsers sessionUsers;
+
     private final SessionUserCount sessionUserCount;
 
-    public Session(SessionImage sessionImage, SessionPeriod sessionPeriod, SessionPrice sessionPrice, SessionState sessionState, SessionType sessionType, SessionUserCount sessionUserCount) {
+    public Session(Long id) {
+        this(id, null, null, null, null, null, null, null, null);
+    }
+
+    public Session(Long id, Long courseId, SessionImage sessionImage, SessionPeriod sessionPeriod, SessionPrice sessionPrice, SessionState sessionState, SessionType sessionType, SessionUserCount sessionUserCount) {
+        this(id, courseId, sessionImage, sessionPeriod, sessionPrice, sessionState, sessionType, new SessionUsers(), sessionUserCount);
+    }
+
+    public Session(Long id, Long courseId, SessionImage sessionImage, SessionPeriod sessionPeriod, SessionPrice sessionPrice, SessionState sessionState, SessionType sessionType, SessionUsers sessionUsers, SessionUserCount sessionUserCount) {
+        this.id = id;
+        this.courseId = courseId;
         this.sessionImage = sessionImage;
         this.sessionPeriod = sessionPeriod;
         this.sessionPrice = sessionPrice;
         this.sessionState = sessionState;
         this.sessionType = sessionType;
+        this.sessionUsers = sessionUsers;
         this.sessionUserCount = sessionUserCount;
     }
 
-    public void register(Payment payment) {
+    public void register(NsUser user, Payment payment) {
         validateState();
         validateType();
         validatePriceEqualPayment(payment);
+        sessionUsers.addUser(user);
         sessionUserCount.plusUserCount();
     }
```

Comment:
> 유저 중복 신청을 막기 위해 Session내 SessionUsers를 만들었습니다. 해당 구조가 올바른지 확신이 들진 않네요..

잘 구현하신 것 같아요 👍 
다만 지금처럼 기능이 추가되거나 변경되면 필드도 추가되고 메서드 내부도 계속 늘어날 것 같은데요
수강신청 객체를 도출해서 관련 책임들을 모두 위임하는 구조도 고려해 보시면 좋을 것 같아요
```suggestion
    public void register(NsUser user, Payment payment) {
        enrollment.enroll(user, payment);
    }
```

Reply:
확실히 위임하니 더 좋네요!! 감사합니다~